src/graph/rg_matrix/rg_matrix_iter.c

```diff
@@ -356,12 +350,7 @@ GrB_Info RG_MatrixTupleIter_detach
 ) {
 	ASSERT(iter != NULL) ;
 
-	if(iter->m_it  != NULL) GrB_free (&iter->m_it) ;
-	if(iter->dp_it != NULL) GrB_free (&iter->dp_it) ;
-
-	iter->A     = NULL ;
-	iter->m_it  = NULL ;
-	iter->dp_it = NULL ;
+	iter->A = NULL ;
```

Comment:
`iter->depleted = true;`
see if there are places where we can avoid checking if `iter->A == NULL`